SignalMessaging/attachments/SignalAttachment.swift

```diff
@@ -160,6 +160,10 @@ public class SignalAttachment: NSObject {
     @objc
     public let dataUTI: String
 
+    // Can be used by views to link this SignalAttachment with an arbitrary source.
+    @objc
+    public var sourceId: String?
```

Comment:
This was a _lot_ more expedient & clean than any other linking mechanism I could come up with.

Reply:
That seems fine, but can we call this field `assetId` instead? or `sourceAssetId`

Reply:
Done.